src/headless/converse-chatboxes.js

```diff
@@ -405,7 +405,7 @@ converse.plugins.add('converse-chatboxes', {
                 }
                 const room_jids = _converse.auto_join_rooms.map(s => _.isObject(s) ? s.jid : s);
                 const auto_join = _converse.auto_join_private_chats.concat(room_jids);
-                if (_converse.singleton && !_.includes(auto_join, attrs.jid)) {
+                if (_converse.singleton && !_.includes(auto_join, attrs.jid) && !_converse.auto_join_on_invite) {                
```

Comment:
Thanks @deleolajide. Please remove the trailing whitespace and also add an entry to the changelog.

Reply:
I have updated changelog. Not sure about white spaces. I made changes directly from github.com website. I can't tell if they are removed or not.